### ISSUE TITLE
False positive for `@typescript-eslint/no-unnecessary-condition` with `export let`

### ISSUE THREAD

**Issue by kevincox** [sveltejs/eslint-plugin-svelte3#203]
```svelte
<script lang="ts">
	export let mode: "foo" | "bar" = "foo";
	mode == "foo";
</script>
```

```
src/Test.svelte
  3:2  warning  Unnecessary conditional, both sides of the expression are literal values  @typescript-eslint/no-unnecessary-condition
```

It seems like typescript doesn't realize that the prop is mutable.

The problem doesn't exist for various other ways of writing this:

``` typescript
export let mode = "foo" as "foo" | "bar";
export let mode = "foo";
export let mode: "foo" | "bar";
```

In all of these cases no warning is emitted.

This occurs for more than just union types. For example it also works with enums and booleans.

---

- @typescript-eslint/eslint-plugin: 5.55.0
- @typescript-eslint/parser: 5.55.0
- eslint: 8.36.0
- eslint-plugin-svelte3: 4.0.0
- svelte: 3.57.0
- typescript: 4.9.5

```javascript
module.exports = {
	extends: [
		"eslint:recommended",
		"plugin:@typescript-eslint/strict",
	],
	parser: "@typescript-eslint/parser",
	parserOptions: {
		project: "tsconfig.json",
		extraFileExtensions: [".svelte"],
	},
	overrides: [{
		files: ["*.svelte"],
		processor: "svelte3/svelte3",
	}],
	plugins: [
		"@typescript-eslint",
		"svelte3",
	],
	settings: {
		"svelte3/typescript": true,
	},
};
```
